src/include/cpu/rpp_cpu_common.hpp

```diff
@@ -75,6 +75,106 @@ inline int fastrand()
+    xorwowSeedStream[2] = seed + 4377713U;
+    xorwowSeedStream[3] = seed + 1648031U;
+    for (int i = 0; i < 4; i++)
+    {
+        xorwowInitialState[i].x[0] = 123456789U + xorwowSeedStream[i];
+        xorwowInitialState[i].x[1] = 362436069U + xorwowSeedStream[i];
+        xorwowInitialState[i].x[2] = 521288629U + xorwowSeedStream[i];
+        xorwowInitialState[i].x[3] = 88675123U + xorwowSeedStream[i];
+        xorwowInitialState[i].x[4] = 5783321U + xorwowSeedStream[i];
+        xorwowInitialState[i].counter = 6615241U + xorwowSeedStream[i];
+    }
+}
+
+inline __m256 rpp_host_rng_xorwow_8_f32_avx(__m256i *pxXorwowStateXParam, __m256i *pxXorwowStateCounterParam)
+{
+    __m256i px362437 = _mm256_set1_epi32(362437);
+    __m256i pxFFFFFFFF = _mm256_set1_epi32(0xFFFFFFFF);
+    __m256i px7FFFFF = _mm256_set1_epi32(0x7FFFFF);
+    __m256i pxExponentFloat = _mm256_set1_epi32(0b111111100000000000000000000000);
+    __m256i pxT = pxXorwowStateXParam[4];
+    __m256i pxS = pxXorwowStateXParam[0];
+    pxXorwowStateXParam[4] = pxXorwowStateXParam[3];
+    pxXorwowStateXParam[3] = pxXorwowStateXParam[2];
+    pxXorwowStateXParam[2] = pxXorwowStateXParam[1];
+    pxXorwowStateXParam[1] = pxS;
+    pxT = _mm256_xor_si256(pxT, _mm256_srli_epi32(pxT, 2));
+    pxT = _mm256_xor_si256(pxT, _mm256_slli_epi32(pxT, 1));
+    pxT = _mm256_xor_si256(pxT, _mm256_xor_si256(pxS, _mm256_slli_epi32(pxS, 4)));
+    pxXorwowStateXParam[0] = pxT;
+    *pxXorwowStateCounterParam = _mm256_and_si256(_mm256_add_epi32(*pxXorwowStateCounterParam, px362437), pxFFFFFFFF);
+    pxS = _mm256_or_si256(pxExponentFloat, _mm256_and_si256(_mm256_add_epi32(pxT, *pxXorwowStateCounterParam), px7FFFFF));
+    return _mm256_sub_ps(*(__m256 *)&pxS, avx_p1);
+}
+
+inline __m128 rpp_host_rng_xorwow_4_f32_sse(__m128i *pxXorwowStateXParam, __m128i *pxXorwowStateCounterParam)
+{
+    __m128i px362437 = _mm_set1_epi32(362437);
+    __m128i pxFFFFFFFF = _mm_set1_epi32(0xFFFFFFFF);
+    __m128i px7FFFFF = _mm_set1_epi32(0x7FFFFF);
+    __m128i pxExponentFloat = _mm_set1_epi32(0b111111100000000000000000000000);
+    __m128i pxT = pxXorwowStateXParam[4];
+    __m128i pxS = pxXorwowStateXParam[0];
+    pxXorwowStateXParam[4] = pxXorwowStateXParam[3];
+    pxXorwowStateXParam[3] = pxXorwowStateXParam[2];
+    pxXorwowStateXParam[2] = pxXorwowStateXParam[1];
+    pxXorwowStateXParam[1] = pxS;
+    pxT = _mm_xor_si128(pxT, _mm_srli_epi32(pxT, 2));
+    pxT = _mm_xor_si128(pxT, _mm_slli_epi32(pxT, 1));
+    pxT = _mm_xor_si128(pxT, _mm_xor_si128(pxS, _mm_slli_epi32(pxS, 4)));
+    pxXorwowStateXParam[0] = pxT;
+    *pxXorwowStateCounterParam = _mm_and_si128(_mm_add_epi32(*pxXorwowStateCounterParam, px362437), pxFFFFFFFF);
+    pxS = _mm_or_si128(pxExponentFloat, _mm_and_si128(_mm_add_epi32(pxT, *pxXorwowStateCounterParam), px7FFFFF));
+    return _mm_sub_ps(*(__m128 *)&pxS, xmm_p1);
+}
+
+inline float rpp_host_rng_xorwow_f32(RpptXorwowState *xorwowState)
+{
+    uint t  = xorwowState->x[4];
```

Comment:
I would like to review this separate. It needs rework

Reply:
x-params reordered in the xorwow state